apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -807,7 +817,7 @@ type perContainerHandler interface {
 	// machine that is hosting the container.
 	// Any errors that are returned from ProcessOneContainer will be turned
 	// into ServerError and handed to SetError
-	ProcessOneContainer(env environs.Environ, callContext context.ProviderCallContext, idx int, host, container *state.Machine) error
+	ProcessOneContainer(env environs.Environ, callContext context.ProviderCallContext, idx int, host, container Machine) error
```

Comment:
I'm happy to see this. Less code outside of state that depends on "state" objects is a great thing.